src/libfetchers/fetchers.cc

```diff
@@ -354,6 +354,8 @@ std::pair<ref<SourceAccessor>, Input> Input::getAccessorUnchecked(ref<Store> sto
 
     if (!accessor->fingerprint)
         accessor->fingerprint = result.getFingerprint(store);
+    else
```

Comment:
Since you note "this is kinda hacky" in the PR body, should we add a FIXME here with some context on why that is and how it could be fixed at some point?